clay.h

```diff
@@ -1436,7 +1395,55 @@ void Clay__CloseElement(void) {
     }
 }
 
-void Clay__OpenElement(void) {
+bool Clay__MemCmp(const char *s1, const char *s2, int32_t length);
+
+#ifdef __x86_64__
+    bool Clay__MemCmp(const char *s1, const char *s2, int32_t length) {
+        for (int32_t i = 0; i < length; i++) {
+            if (s1[i] != s2[i]) {
+                return false;
+            }
+        }
+        return true;
+    }
+#elif defined(__aarch64__)
+    bool Clay__MemCmp(const char *s1, const char *s2, int32_t length) {
+        while (length >= 16) {
+            uint8x16_t v1 = vld1q_u8((const uint8_t *)s1);
+            uint8x16_t v2 = vld1q_u8((const uint8_t *)s2);
+
+            // Compare vectors
+            uint8x16_t cmp = vceqq_u8(v1, v2);
+            uint64_t mask = vgetq_lane_u64(vreinterpretq_u64_u8(cmp), 0) &
+                            vgetq_lane_u64(vreinterpretq_u64_u8(cmp), 1);
+
+            if (mask != UINT64_MAX) { // If there's a difference
+                for (int32_t i = 0; i < 16; i++) {
+                    if (s1[i] != s2[i]) {
+                        return false;
+                    }
+                }
+            }
```

Comment:
I think this can be simplified down to (untested)
```C
if (vminvq_u32(mask) != 0xFFFFFFFF) { // If any element is not equal
    return false;
}
```

Reply:
Another approach is to use 
```
const uint8x8_t res = vshrn_n_u16(mask, 4);
const uint64_t matches = vget_lane_u64(vreinterpret_u64_u8(res), 0);
```
this gives you a `u64` value where each nibble is 4 bit representation of the compare mask. ref https://community.arm.com/arm-community-blogs/b/servers-and-cloud-computing-blog/posts/porting-x86-vector-bitmask-optimizations-to-arm-neon

Reply:
Awesome, thanks for the tips!